lib/compress/zstdmt_compress.c

```diff
@@ -142,36 +142,42 @@ static size_t ZSTDMT_sizeof_bufferPool(ZSTDMT_bufferPool* bufPool)
 
 static void ZSTDMT_setBufferSize(ZSTDMT_bufferPool* bufPool, size_t bSize)
 {
+    ZSTD_pthread_mutex_lock(&bufPool->poolMutex);
```

Comment:
I'm okay with acquiring the lock, but it isn't currently necessary right?

Reply:
Yes, right.
It's a bit overcautious, but should not impact performance.

I briefly suspected `bSize` initialisation as a possible source for a bug I was hunting, adding the lock helped removing that possibility.